File: 70 Climbing Stairs/solution.cpp
class Solution {
public:
    int climbStairs(int n) {
        int prev = 1,cur = 2,next = 0;
        if (n<3)
            next = n;
        else{
            for (int i=3;i<=n;i++){
                next = prev+cur;
                prev = cur;
                cur = next;
            }
        }
        return next;
    }
};

File: 70 Climbing Stairs/solution.py
class Solution:
    # @param {integer} n
    # @return {integer}
    def climbStairs(self, n):
        if n<3:
            return n
        result = []
        for i in range(n+1):
            if i==0 or i==1 or i==2:
                result.append(i)
            else:
                result.append(result[i-1]+result[i-2])
        return result[-1]
        